liblangutil/Exceptions.h

```diff
@@ -238,35 +247,72 @@ class Error: virtual public util::Exception
 
 	static std::string formatErrorSeverity(Severity _severity)
 	{
-		if (_severity == Severity::Info)
+		switch (_severity)
+		{
+		case Severity::Info:
 			return "Info";
-		if (_severity == Severity::Warning)
+		case Severity::Warning:
 			return "Warning";
-		solAssert(isError(_severity), "");
-		return "Error";
+		default:
+			solAssert(isError(_severity));
+			return "Error";
+		}
 	}
 
-	static std::string formatErrorSeverityLowercase(Severity _severity)
+	static std::string formatErrorType(Type _type)
 	{
-		switch (_severity)
+		switch (_type)
 		{
-		case Severity::Info:
-			return "info";
-		case Severity::Warning:
-			return "warning";
-		case Severity::Error:
-			solAssert(isError(_severity), "");
-			return "error";
+		case Type::IOError:
+			return "IOError";
+		case Type::FatalError:
+			return "FatalError";
+		case Type::JSONError:
+			return "JSONError";
+		case Type::InternalCompilerError:
+			return "InternalCompilerError";
+		case Type::CompilerError:
+			return "CompilerError";
+		case Type::Exception:
+			return "Exception";
+		case Type::CodeGenerationError:
+			return "CodeGenerationError";
+		case Type::DeclarationError:
+			return "DeclarationError";
+		case Type::DocstringParsingError:
+			return "DocstringParsingError";
+		case Type::Info:
+			return "Info";
+		case Type::ParserError:
+			return "ParserError";
+		case Type::SyntaxError:
+			return "SyntaxError";
+		case Type::TypeError:
+			return "TypeError";
+		case Type::UnimplementedFeatureError:
+			return "UnimplementedFeatureError";
+		case Type::YulException:
+			return "YulException";
+		case Type::SMTLogicException:
+			return "SMTLogicException";
+		default:
```

Comment:
```suggestion
		case Type::Warning:
```
I'd also put `Warning` and `Info` next to each other.